libr/bin/format/mach0/mach0.c

```diff
@@ -4267,7 +4272,7 @@ RList *MACH0_(mach_fields)(RBinFile *bf) {
 			ut32 nsects = r_buf_read_le32_at (buf, addr + (is64 ? 64 : 48));
 			ut64 off = is64 ? 72 : 56;
 			int j = 0;
-			while (off < lcSize && nsects--) {
+			while (off < lcSize && (addr + off) < length && nsects-- > 0) {
```

Comment:
please use a for if you need to decrement. makes the code more readable and prevents weird usecases.